### ISSUE TITLE
TPM plugin for this would be cool

### ISSUE THREAD

**Issue by waynr** [erikw/tmux-powerline#189]
It'd be nice to install/manage this the same way I am managing other tmux extensions.

**Comment by SimenB** [comment]
:+1:

**Comment by erikw** [comment]
Note: Here is the doc for how to create a TPM plugin
 https://github.com/tmux-plugins/tpm/blob/master/docs/how_to_create_plugin.md

**Comment by TheSast** [comment]
bump.
This would be really useful, currently tpm deletes tmux-powerline if it is placed in the plugins folder upon pressing `<prefix> + M-u`.
Also tpm plugin would make it easier to order plugins correctly (e.g. tmux-continuum needs to be placed after any status themes to work correctly)

**Comment by erikw** [comment]
**TL;DR tmux-powerline can now be installed as a tpm plugin**

You're in luck,  I created some weeks ago a tpm plugin [tmux-dark-notify](https://github.com/erikw/tmux-dark-notify) so I know how to make tpm plugins now.

🎉 Today I converted tmux-powerline to be installable as a tpm plugin! 🥳 
(8 years after the issue was opened lol, better late than never)

_It Works On My Machine™_
It would be great if one of you could install tmux-powerline via tpm to make sure it's working properly on other than my computer :)

I'd recommend to
1. start from scratch that is remove any previous tmux-powerline setup from `tmux.conf` and then proceed with the instructions at https://github.com/erikw/tmux-powerline#installation-the-tpm-way-recommended
2. At the configuration stage, you can reuse your existing tmux-powerline configuration and themes.
  * Not that the old way of installing tmux-powerline instructed you to set the tmux options`status-left-length` and `status-right-length` yourself in your tmux.conf
  * Remove these from your tmux.conf, as the length is now configured directly from your tmux-powerline config, to make the tpm installation flow as easy as possbile (just add the plugin to tpm!)
  * The left/right length are now set in tmux-powerline config in the variables `TMUX_POWERLINE_STATUS_LEFT_LENGTH` and same for the right.
  * Remember you can always run the `generate_rc.sh` to generate a new default configuration, and merge this with your old config.
  
  
💬 I'm happy for some feedback, if the tpm installation worked :) 
  
  
 CC those that seemed interested in having this as a tpm plugin:
@waynr @SimenB @TheSast @albttx @akz92 @joebalancio @girlandhercode @palcu @wfxr @soulomoon @wpcarro @Maxattax97 @0x450x6c @wxfr @ProfMoo

**Comment by TheSast** [comment]
First thing I'll check tomorrow, will provide feedback as soon as I can.
Now that we got tpm support I'll also be able to try and impl a better way to set the rc filepath and the muting bindings (maybe the status length too but that is easily overridable)
